geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMemberData.java

```diff
@@ -220,12 +220,16 @@ public InetAddress getInetAddress() {
   @Override
 
   public short getVersionOrdinal() {
-    return this.versionObj.ordinal();
+    return versionOrdinal;
   }
 
   @Override
   public Version getVersion() {
-    return versionObj;
+    try {
+      return Version.fromOrdinal(versionOrdinal);
+    } catch (final UnsupportedSerializationVersionException e) {
```

Comment:
**This right here** is a big fat problem. We are silently throwing away information when the version ordinal represents an unknown version. One approach is to let the exception escape (as a checked exception.) But that entails fixing the 53 callers of this method. An alternative would be to "tunnel" the checked exception through an unchecked one.